### PR TITLE
Template

### LINKED ISSUE #1: Template project for aquality-appium-mobile-java

### PR DESCRIPTION
**Pull request by paveliam**

### CODE REVIEW

**Review by DmitryBogatko — APPROVED** [review]

**Review by knysh — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,2 +1,40 @@
...
+  - **src/main/resources/**: resource files such as configurations and test data
+- **aquality-appium-mobile-template-cucumber** - Cucumber implementation of the tests
+  - **features/**: Gherkin feature files with test scenarios
+  - **hooks/**: Cucumber [hooks](https://specflow.org/documentation/Hooks/)
+  - **stepdefinitions/**: step definition classes
+  - **transformations/**: Cucumber [data transformations](https://cucumber.io/docs/cucumber/configuration/)
+  - **objectfactory/**: [DI container](https://cucumber.io/docs/cucumber/state/#how-to-use-di) configuration
+
+### Configuration
+[settings.json](https://github.com/aquality-automation/aquality-appium-mobile-java-template/blob/master/aquality-appium-mobile-template/src/main/resources/settings.json) file contains settings of Aquality Appium Mobile library. Additional information you can find [here](https://github.com/aquality-automation/aquality-selenium-java/wiki/Overview-(English)).
```
Comment (mialeska) [inline README.md]: Is the link to https://github.com/aquality-automation/aquality-selenium-java/wiki/Overview-(English) correct?

File: aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/CustomObjectFactory.java [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import cucumber.api.guice.CucumberModules;
+import cucumber.runtime.java.guice.ScenarioScope;
+import io.cucumber.core.backend.ObjectFactory;
+
+public class CustomObjectFactory implements ObjectFactory {
+
+    private Injector injector;
+
+    public CustomObjectFactory() {
+        this.injector = Guice.createInjector(CucumberModules.createScenarioModule(), new ServiceModule());
```
Comment (mialeska) [inline aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/CustomObjectFactory.java]: Should we just add MobileModule additionally here?
this way we can resolve all aquality services

File: aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/ServiceModule.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+import aquality.appium.mobile.application.AqualityServices;
+import aquality.appium.mobile.screens.screenfactory.IScreenFactory;
+import aquality.appium.mobile.template.utilities.ScreenshotProvider;
+import com.google.inject.AbstractModule;
+
+public class ServiceModule extends AbstractModule {
+
+    @Override
+    protected void configure() {
+        bind(IScreenFactory.class).to(AqualityServices.getScreenFactory().getClass());
```
Comment (mialeska) [inline aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/ServiceModule.java]: probably we should bind to instance here

File: aquality-appium-mobile-template/src/main/resources/settings.json [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+  },
+  "retry": {
+    "number": 2,
+    "pollingInterval": 300
+  },
+  "logger": {
+    "language": "en"
+  },
+  "elementCache": {
+    "isEnabled": false
```
Comment (mialeska) [inline aquality-appium-mobile-template/src/main/resources/settings.json]: :_(

File: aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/CustomObjectFactory.java (hunk 2 above)
Comment (paveliam) [reply aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/CustomObjectFactory.java]: Do we really need it? Actually, it'd be better to use DI in all our templates. Let's discuss.

File: README.md (hunk 1 above)
Comment (paveliam) [reply README.md]: The concepts of settings is OK, but we don't have docs in appium library.

File: aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/ServiceModule.java (hunk 3 above)
Comment (paveliam) [reply aquality-appium-mobile-template-cucumber/src/test/java/aquality/appium/mobile/template/cucumber/objectfactory/ServiceModule.java]: Why? AqualityServices.getScreenFactory() - it is an instance. We have to bind to instance when we don't have interfaces, isn't it?

**Review by mialeska — APPROVED** [review]

**Pull request merged**